rust/kernel/net/ethernet.rs

```diff
@@ -0,0 +1,435 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Ethernet Module
+//!
+//! C headers: [`include/uapi/linux/if_ether.h`](../../../../include/uapi/linux/if_ether.h)
+
+use crate::bindings;
+
+// Some constants
+// These were taken from `include/uapi/linux/if_ether.h`
+/// IEEE 802.3 Ethernet magic constants
+///
+/// Taken from the original tree at `include/uapi/linux/if_ether.h`.
+#[repr(u32)]
+pub enum Lengths {
```

Comment:
I don't think these should be in an enum. There is no reason that the different constants would not have the same value (which an enum doesn't allow), nor is there any reason to store it as enum as opposed to immediately casting it back to the discriminator.

Reply:
Wouldn't it also make more sense to store them as `usize`? I originally avoided it because `usize`'s size is architecture dependent and I don't want that as a side effect, but idiomatically it makes more sense. For now I will rewrite all of those as `pub const u32`.

Reply:
I think `usize` makes most sense. It seems like most of the time it is either used as length of an array or used in pointer arithmetic. Both of which require `usize` in rust.

Reply:
Agreed. I forgot Rust would not even build in 8 bit, so I changed it. It's in the latest force-push. (marking this resolved)

Reply:
I don't think we need to care for 16-bit either.

Reply:
Given that in kernel `int` is `i32` and `long` is `isize` and `sizeof(int) <= sizeof(long)`, we would never have to worry about 16-bit `usize`.

Reply:
Indeed, nice proof!